worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java

```diff
@@ -26,43 +29,51 @@
 import com.hpe.caf.worker.document.changelog.MutableDocument;
 import com.hpe.caf.worker.document.config.DocumentWorkerConfiguration;
 import com.hpe.caf.worker.document.exceptions.InvalidChangeLogException;
+import com.hpe.caf.worker.document.exceptions.PostProcessingFailedException;
 import com.hpe.caf.worker.document.impl.ApplicationImpl;
+import com.hpe.caf.worker.document.model.ResponseOptions;
 import com.hpe.caf.worker.document.output.ChangeLogBuilder;
 import com.hpe.caf.worker.document.util.ListFunctions;
 import com.hpe.caf.worker.document.views.ReadOnlyDocument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
-import javax.annotation.Nonnull;
 
 public final class DocumentTask extends AbstractTask
 {
+    private static final Logger LOG = LoggerFactory.getLogger(DocumentTask.class);
     private final DocumentWorkerDocumentTask documentTask;
+    private final DocumentPostProcessor postProcessor;
 
     @Nonnull
     public static DocumentTask create(
-        final ApplicationImpl application,
-        final WorkerTaskData workerTask,
-        final DocumentWorkerDocumentTask documentTask
-    ) throws InvalidChangeLogException
+            final ApplicationImpl application,
+            final WorkerTaskData workerTask,
+            final DocumentWorkerDocumentTask documentTask,
+            final DocumentPostProcessorFactory postProcessorFactory) throws InvalidChangeLogException, TaskRejectedException
     {
         Objects.requireNonNull(documentTask);
 
-        return new DocumentTask(application, workerTask, documentTask);
+        return new DocumentTask(application, workerTask, documentTask, postProcessorFactory);
     }
 
     private DocumentTask(
-        final ApplicationImpl application,
-        final WorkerTaskData workerTask,
-        final DocumentWorkerDocumentTask documentTask
-    ) throws InvalidChangeLogException
+            final ApplicationImpl application,
+            final WorkerTaskData workerTask,
+            final DocumentWorkerDocumentTask documentTask,
+            DocumentPostProcessorFactory postProcessorFactory) throws InvalidChangeLogException, TaskRejectedException
```

Comment:
I change the other class (app impl), but left this because I didn't want to update the interface. If you think it should be on the interface for Application, I'll change it.